### PR TITLE
CASMTRIAGE-7232 CSM 1.5 kdump hotfix

### PR DESCRIPTION
**Pull request by rustydb**
This adds a hotfix for kdump in 1.5.

Requires: https://github.com/Cray-HPE/metal-ipxe/pull/83

In some version of 1.5, namely 1.5.2, kdump will emit the following error during system startup:
```
Cannot reserve 256MB crashkernel low memory, please try smaller size.
```
The result is no core dump is taken during a system crash.

The error is new as of a minor version bump of kdump, but the changelogs were inconclusive.

The fix was to simply set a `low` memory bound for kdump. After setting this on the commandline the system is able to take core dumps again.

### Tests

Tested on vshasta (scanlan) with CSM v1.5.2:

- `hotfix-fresh-install.sh` was ran via the pit. Using `zypper`, the newer metal-ipxe package was discovered:
    ```bash
    pit:/etc/zypp/repos.d # cd
    pit:~ # cat /etc/zypp/repos.d/noos.repo
    [noos]
    enabled=1
    autorefresh=1
    baseurl=dir:/var/www/ephemeral/csm-1.5.2/rpm/cray/csm/noos
    gpgcheck=0
    pit:~ # zypper -q up --details metal-ipxe

    The following package is going to be upgraded:
      metal-ipxe  2.4.7-1 -> 2.4.102-1  noarch  noos  Cray Inc.

    The following package has no support information from its vendor:
      metal-ipxe  2.4.102-1  noarch  noos  Cray Inc.

    1 package to upgrade.
    Overall download size: 606.4 KiB. Already cached: 0 B. After the operation, additional 79.5 KiB will be used.
    Continue? [y/n/v/...? shows all options] (y): ^C
    Trying to exit gracefully...
    ```
- `hotfix-running-system.sh` was ran against the NCNs via ncn-m001. The patched BSS files all contained the correct `crashkernel` arguments.

> ***NOTE*** The vshasta system I used, scanlan, was missing its `/metal/recovery/boot/grub2/grub.cfg` file on the NCNs. There appears to be a bug in vShasta, perhaps for just 1.5, where `grub.cfg` fails to create. I've tested the `grub.cfg` portion of my PR already, so I am going to advocate for merging as-is. However, I added a new conditional that will fail the script if turns out to be missing `grub.cfg`.

### CODE REVIEW

**Comment by rustydb** [comment]
I need to update this to blacklist additional modules in `/etc/sysconfig/kdump` in the runtime script.

File: csm-1.5/CASMTRIAGE-7232-kdump/hotfix-running-system.sh [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+LOG_DIR=/var/log/hotfix-kdump
+CURRENT_LOG_DIR="${LOG_DIR}/$(date '+%Y-%m-%d_%H:%M:%S')"
+mkdir -p "${CURRENT_LOG_DIR}"
+exec 19> "${CURRENT_LOG_DIR}/patch.xtrace"
+export BASH_XTRACEFD="19"
+
+function usage {
+
+  cat << EOF
+Updates BSS the crashkernel boot parameter in BSS and on disk bootloaders
```
Comment (jpdavis-prof) [inline csm-1.5/CASMTRIAGE-7232-kdump/hotfix-running-system.sh:35]: Worded a little weird.

File: csm-1.5/CASMTRIAGE-7232-kdump/hotfix-running-system.sh [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  exit 1
+fi
+echo 'Done'
+echo "Found: [${#EXPECTED_NCNS[@]}] NCN(s)"
+if ! update-disk-bootloaders "${EXPECTED_NCNS[@]}"; then
+  echo >&2 'Failed to update one or more disk bootloaders. Aborting.'
+  exit 1
+fi
+
+function update-bss() {
```
Comment (jpdavis-prof) [inline csm-1.5/CASMTRIAGE-7232-kdump/hotfix-running-system.sh:143]: Don't need parens for consistency.

**Review by jpdavis-prof — APPROVED** [review]

File: csm-1.5/CASMTRIAGE-6796-csm-pre-install-metal-ipxe/rpm/index.yaml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 https://artifactory.algol60.net/artifactory/csm-rpms/hpe/stable/noos/:
   rpms:
-    - metal-ipxe-2.4.8-1.noarch
+    - metal-ipxe-2.4.101-1.noarch
```
Comment (rustydb) [inline csm-1.5/CASMTRIAGE-6796-csm-pre-install-metal-ipxe/rpm/index.yaml:3]: This mismatches the other rpm index yaml, I'll double check if that's correct or not. Pretty sure this should be 102.

File: csm-1.5/CASMTRIAGE-7232-kdump/hotfix-running-system.sh [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      echo "Done"
+      printf "Patching disk bootloader ... "
+
+      sed -i -E "s/crashkernel=[0-9]+[a-zA-Z]?//g" "$BOOTRAID/boot/grub2/grub.cfg"
+
+      sed -i -E '\''s/(crashkernel=)[0-9]+[a-zA-Z]/\1512M,high \172M,low/'\'' "$BOOTRAID/boot/grub2/grub.cfg"
+      echo "Done"
+    fi
+  ' 2>/dev/null | dshbak -c ; then
+      echo >&2 'Failed! to apply the updated crashkernel bootparemeters to one or more nodes.'
```
Comment (rustydb) [inline csm-1.5/CASMTRIAGE-7232-kdump/hotfix-running-system.sh:124]: Wording.

File: csm-1.5/CASMTRIAGE-6796-csm-pre-install-metal-ipxe/rpm/index.yaml (hunk 3 above)
Comment (rustydb) [reply csm-1.5/CASMTRIAGE-6796-csm-pre-install-metal-ipxe/rpm/index.yaml:3]: This is right. We want to leave this at `2.4.101` and direct users to the newer hotfix.

**Review by manderson-hpe — APPROVED** [review]

**Review by jpdavis-prof — APPROVED** [review]

**Pull request merged**